pyanaconda/users.py

```diff
@@ -161,6 +164,20 @@ def validatePassword(pw, user="root", settings=None):
 
     return (valid, strength, message)
 
+def check_name(in_name):
+    ign_dir = {}
+    for inc in os.listdir("/") + ["root", "home", "daemon"]:
+        ign_dir.update({inc: "Cannot set user name, reserved for system: "})
+    ign_dir.update({r"^[\d]": "User name cannot start with a digit: "})
+    ign_dir.update({r"[^A-Za-z0-9\s]": "User name cannot contain any non-alphanumerical character: "})
+
+    for inc in ign_dir:
+        match = regsearch(inc, in_name)
+        if match:
+            return (False, ign_dir[inc] + "'%s'" % match.group())
```

Comment:
What is the point of returning the match here when it is later just put into this: "You have provided an invalid user name: %s\n"? E.g. for ``"123my_username"`` this would put ``"1"`` there with this result shown to the user:
```
You have provided an invalid user name: 1
```